src/App.tsx

```diff
@@ -1,61 +1,117 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList';
+import todosFromServer from './api/todos';
+
+const getUserById = (userId:number) => {
+  return usersFromServer.find((user) => userId === user.id);
+};
+
+const todosFormat = todosFromServer.map((todo) => {
+  return {
+    ...todo,
+    user: getUserById(todo.userId) || null,
+  };
+});
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userInfo, setUserInfo] = useState('');
+  const [todos, setTodos] = useState(todosFormat);
+  const [error, setError] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setError(false);
+    setTitle(event.currentTarget.value);
+  };
+
+  const handleChangeName = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setErrorUser(false);
+    setUserInfo(event.currentTarget.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const idUsers = todos.map((person) => person.id);
+    const largestId = Math.max(...idUsers);
+
+    const user = getUserById(Number(userInfo));
+
+    if (title.trim() === '') {
+      setError(true);
+    }
+
+    if (userInfo === '') {
+      setErrorUser(true);
+    }
+
+    if (!(title === '' || userInfo === '')) {
+      setTodos((prev): any => {
```

Comment:
I'm not sure if you need to use type here. But if you need, try to specify type (Todo[] or smth else)